sympy/core/function.py

```diff
@@ -673,6 +673,13 @@ def _eval_as_leading_term(self, x):
         else:
             return self.func(*args)
 
+    def _sage_(self):
+        import sage.all as sage
+        fname = self.func.__name__
+        func = getattr(sage, fname)
```

Comment:
Perhaps, you can add some private attribute to keep the "right" sage name?  This is an alternative for overriding _sage_ methods below.